### PR TITLE
Drag and Drop improvements implementation.

### PR DESCRIPTION
**Pull request by skrustev**
Related #5319

New Functionalities:
- [x] [Dragging without ghost](https://github.com/IgniteUI/igniteui-angular/wiki/Drag&Drop-Directives-Specification#dragging-without-ghost)
- [x] [Ghost template](https://github.com/IgniteUI/igniteui-angular/wiki/Drag&Drop-Directives-Specification#dragging-with-ghost-element)
- [x] [Dragging using a handle](https://github.com/IgniteUI/igniteui-angular/wiki/Drag&Drop-Directives-Specification#dragging-using-a-handle)
- [x] [Linking of drag and drop elements](https://github.com/IgniteUI/igniteui-angular/wiki/Drag&Drop-Directives-Specification#linking-drag-to-drop-element)

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [x] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts [hunk 1]
```diff
@@ -893,20 +1249,38 @@ export class IgxDropDirective implements OnInit, OnDestroy {
...
     public data: any;
 
+    /**
+     * An @Input property that provide a way for igxDrag and igxDrop to be linked through channels.
+     * It accepts single value or an array of values and evaluates then using strict equality.
+     * ```html
+     * <div igxDrag [dragChannel]="'odd'">
+     *         <span>95</span>
+     * </div>
+     * <div igxDrag [dropChannel]="['odd', 'irrational']">
```
Comment (dkamburov) [inline projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts]: This should be `igxDrop`

File: projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts [hunk 2]
```diff
@@ -151,52 +157,104 @@ export interface IDragStartEventArgs extends IDragBaseEventArgs {
...
     public dragTolerance = 5;
 
     /**
+     * An @Input property that provide a way for igxDrag and igxDrop to be linked through channels.
+     * It accepts single value or an array of values and evaluates then using strict equality.
+     * ```html
+     * <div igxDrag [dragChannel]="'odd'">
+     *         <span>95</span>
+     * </div>
+     * <div igxDrag [dropChannel]="['odd', 'irrational']">
```
Comment (dkamburov) [inline projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts]: This should be `igxDrop`

File: projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts [hunk 3]
```diff
@@ -151,52 +157,104 @@ export interface IDragStartEventArgs extends IDragBaseEventArgs {
     cancel: boolean;
 }
 
+export interface IDragLocation {
+    pageX: number;
+    pageY: number;
+}
+
+export interface IDragCustomAnimationArgs {
```
Comment (dkamburov) [inline projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts]: I can't see the usage of this interface `IDragCustomAnimationArgs`. The spec states that it should used by `animateToOrigin` and `animateTo`, but those methods are not implemented.

**Comment by MayaKirova** [comment]
igxDragHandle cannot be applied to multiple elements inside the igxDrag. When multiple handles are applied only the first one works.

**Pull request merged**